include/operations/SymbolicOperation.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "Definitions.hpp"
+#include "Expression.hpp"
+#include "StandardOperation.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <optional>
+#include <unordered_set>
+#include <variant>
+
+namespace qc {
+    // Overload pattern for std::visit
+    template<typename... Ts>
+    struct Overload: Ts... {
+        using Ts::operator()...;
+    };
+    template<class... Ts>
+    Overload(Ts...) -> Overload<Ts...>;
+
+    class SymbolicOperation final: public StandardOperation {
+    protected:
+        std::array<std::optional<Symbolic>, MAX_PARAMETERS> symbolicParameter{{std::nullopt, std::nullopt, std::nullopt}};
```

Comment:
I might be wrong, but isn't `std::nullopt` the default anyway?
Then this can just be 
```suggestion
        std::array<std::optional<Symbolic>, MAX_PARAMETERS> symbolicParameter{};
```

(which also wouldn't break if `MAX_PARAMETERS` were to change)